rnd/market/Dockerfile

```diff
@@ -26,31 +19,45 @@ ENV POETRY_VERSION=1.8.3 \
     PATH="$POETRY_HOME/bin:$PATH"
 RUN pip3 install poetry
 
-FROM server_base AS server_dependencies
+# Copy and install dependencies
+COPY rnd/autogpt_libs /app/rnd/autogpt_libs
+COPY rnd/market/poetry.lock rnd/market/pyproject.toml /app/rnd/market/
+WORKDIR /app/rnd/market
+RUN poetry config virtualenvs.create false \
+    && poetry install --no-interaction --no-ansi
+
+# Generate Prisma client
+COPY rnd/market /app/rnd/market
+RUN poetry config virtualenvs.create false && poetry run prisma generate
+
+FROM python:3.11-slim-buster AS server_dependencies
+
+WORKDIR /app
+
+# Copy only necessary files from builder
+COPY --from=builder /app /app
+COPY --from=builder /usr/local/lib/python3.11 /usr/local/lib/python3.11
+COPY --from=builder /usr/local/bin /usr/local/bin
+
+ENV PATH="/app/.venv/bin:$PATH"
 
-RUN mkdir -p /app/autogpt
-RUN mkdir -p /app/forge
 RUN mkdir -p /app/rnd/autogpt_libs
 RUN mkdir -p /app/rnd/market
+
 COPY rnd/autogpt_libs /app/rnd/autogpt_libs
 
 COPY rnd/market/poetry.lock rnd/market/pyproject.toml /app/rnd/market/
 
 WORKDIR /app/rnd/market
 
-RUN poetry install --no-interaction --no-ansi
-
 FROM server_dependencies AS server_prisma
 
-# Need the market/utils/partial_types.py
 COPY rnd/market /app/rnd/market
-
-COPY rnd/market/schema.prisma ./
-RUN poetry run prisma generate
+RUN poetry config virtualenvs.create false && poetry run prisma generate
 
 FROM server_prisma AS server
 
-ENV PORT=8005
 ENV DATABASE_URL=""
+ENV PORT=8000
 
-CMD ["poetry", "run", "app"]
+CMD ["uvicorn", "market.app:app"]
```

Comment:
?

Reply:
run app calls a python scrip that runs uvicorn in a subprocess. I just skip the middle man here